### ISSUE TITLE
ddr-import errors

### ISSUE THREAD

**Issue by gjost** [densho/ddr-cmdln#41]
TL;DR, there is an exception when trying to import only certain entities into ddr repos. The attached stacktrace shows the error I'm getting in all cases when the import fails.

I have test clones of both ddr-manz-1 and ddr-densho-1000 on the qumulo at: `/media/qnfs/kinkura/temp/201706-ddrimport/test`

Some entities will import successfully. I was able to import all of the interview entities in `/media/qnfs/kinkura/temp/201706-ddrimport/ddr-interviews/ddr-manz-1-interviews.csv`; but several of the interviews in `/media/qnfs/kinkura/temp/201706-ddrimport/ddr-interviews/ddr-densho-1000-interviews.csv` fail. Some do work, however; for example, ddr-densho-1000-439. See: `/media/qnfs/kinkura/temp/201706-ddrimport/ddr-segments/ddr-densho-1000-interviews-test.csv`

The same error occurred when trying to import the segments for ddr-manz-1-167 (`/media/qnfs/kinkura/temp/201706-ddrimport/ddr-segments/ddr-manz-1-167-segments.csv`).

A few things I've already tried with no difference in behavior:

- Running `ddr-import` on brand-new clones of ddr-manz-1 and ddr-densho-1000
- Running against clones on local VM storage rather than the qumulo
- Running on a different VM -- the same set of entities failed on both kinkura and maunakea

I also tried mixing entity data that worked with entity data that did not in the same import csv file, just to make sure there wasn't anything wrong with the particular file itself. See: `/media/qnfs/kinkura/temp/201706-ddrimport/ddr-segments/ddr-manz-1-167-segments-test.csv`

I've examined the rows in the csvs that work (i.e., ddr-densho-1000-439 vs. ddr-densho-1000-441 in `ddr-densho-1000-interviews.csv`) and couldn't find anything suspicious such as high-bit chars, extra columns, unterminated escapes, etc.

At this point, I'm not sure what else to try. Looks like it might have something to do with the `identifier.py` code that determines whether the entity already exists; but I'm not certain.

```
(ddrlocal) ddr@maunakea:/media/qnfs/kinkura/temp/201706-ddrimport/test$ ddr-import entity ../ddr-segments/ddr-manz-1-167-segments-test.csv ./ddr-manz-1
2017-06-23 11:03:43,647 DEBUG    <DDR.identifier.Identifier collection:ddr-manz-1>
2017-06-23 11:03:43,648 DEBUG    /media/qnfs/kinkura/temp/201706-ddrimport/test/ddr-manz-1
2017-06-23 11:03:43,648 INFO     ------------------------------------------------------------------------
2017-06-23 11:03:43,648 INFO     batch import entity
2017-06-23 11:03:43,651 INFO     <git.Repo "/media/qnfs/kinkura/temp/201706-ddrimport/test/ddr-manz-1/.git">
2017-06-23 11:03:43,652 INFO     Reading /media/qnfs/kinkura/temp/201706-ddrimport/ddr-segments/ddr-manz-1-167-segments-test.csv
2017-06-23 11:03:43,655 INFO     16 rows
2017-06-23 11:03:43,656 INFO     - - - - - - - - - - - - - - - - - - - - - - - -
2017-06-23 11:03:43,656 INFO     Importing
2017-06-23 11:03:43,656 INFO     1/16 - ddr-manz-1-158-1
2017-06-23 11:03:43,754 DEBUG    | <DDR.identifier.Identifier segment:ddr-manz-1-158-1> (0:00:00.097908)
2017-06-23 11:03:43,755 INFO     2/16 - ddr-manz-1-167-12
Traceback (most recent call last):
  File "/usr/local/src/ddr-local/venv/ddrlocal/bin/ddr-import", line 4, in <module>
    __import__('pkg_resources').run_script('ddr-cmdln==0.9.4b0', 'ddr-import')
  File "/usr/local/src/ddr-local/venv/ddrlocal/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 739, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/src/ddr-local/venv/ddrlocal/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1500, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/src/ddr-local/venv/ddrlocal/lib/python2.7/site-packages/ddr_cmdln-0.9.4b0-py2.7.egg/EGG-INFO/scripts/ddr-import", line 228, in <module>
    main()
  File "/usr/local/src/ddr-local/venv/ddrlocal/lib/python2.7/site-packages/ddr_cmdln-0.9.4b0-py2.7.egg/EGG-INFO/scripts/ddr-import", line 190, in main
    args.dryrun
  File "/usr/local/src/ddr-local/venv/ddrlocal/local/lib/python2.7/site-packages/ddr_cmdln-0.9.4b0-py2.7.egg/DDR/batch.py", line 525, in import_entities
    entity = eidentifier.object()
  File "/usr/local/src/ddr-local/venv/ddrlocal/local/lib/python2.7/site-packages/ddr_cmdln-0.9.4b0-py2.7.egg/DDR/identifier.py", line 1047, in object
    return self.object_class(mappings).from_identifier(self)
  File "/usr/local/src/ddr-local/venv/ddrlocal/local/lib/python2.7/site-packages/ddr_cmdln-0.9.4b0-py2.7.egg/DDR/models/__init__.py", line 1351, in from_identifier
    return from_json(Entity, identifier.path_abs('json'), identifier)
  File "/usr/local/src/ddr-local/venv/ddrlocal/local/lib/python2.7/site-packages/ddr_cmdln-0.9.4b0-py2.7.egg/DDR/models/__init__.py", line 366, in from_json
    document.load_json(fileio.read_text(json_path))
  File "/usr/local/src/ddr-local/venv/ddrlocal/local/lib/python2.7/site-packages/ddr_cmdln-0.9.4b0-py2.7.egg/DDR/fileio.py", line 16, in read_text
    raise IOError('File is missing or unreadable: %s' % path)
IOError: File is missing or unreadable: /media/qnfs/kinkura/temp/201706-ddrimport/test/ddr-manz-1/files/ddr-manz-1-167/files/ddr-manz-1-167-12/entity.json
```

**Comment by gjost** [comment]
> At this point, I'm not sure what else to try. Looks like it might have something to do with the identifier.py code that determines whether the entity already exists; but I'm not certain.

Yep the was exactly what was going on.  The intent of the code was to try to load an existing file (and object) but make a new object if there was no file.  It's been fixed in 4ce6a00.

I tested it with these import files after the update and they now run without incident.
```
ddr-import entity /media/qnfs/kinkura/temp/201706-ddrimport/ddr-interviews/ddr-densho-1000-interviews.csv /media/qnfs/kinkura/temp/201706-ddrimport/test/ddr-densho-1000/
$ ddr-import entity /media/qnfs/kinkura/temp/201706-ddrimport/ddr-segments/ddr-manz-1-167-segments.csv /media/qnfs/kinkura/temp/201706-ddrimport/test/ddr-manz-1/
```
